tests/PhpSpreadsheetTests/Writer/Html/MemoryDrawingOffsetTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace PhpOffice\PhpSpreadsheetTests\Writer\Html;
+
+use Exception;
+use PhpOffice\PhpSpreadsheet\Spreadsheet;
+use PhpOffice\PhpSpreadsheet\Worksheet\MemoryDrawing;
+use PhpOffice\PhpSpreadsheet\Writer\Html;
+use PHPUnit\Framework\TestCase;
+
+class MemoryDrawingOffsetTest extends TestCase
+{
+    /**
+     * @dataProvider dataProvider
+     */
+    public function testMemoryDrawingOffset(int $w, int $h, int $x, int $y): void
+    {
+        $spreadsheet = new Spreadsheet();
+        $sheet = $spreadsheet->getActiveSheet();
+        $drawing = new MemoryDrawing();
+        if (($image = file_get_contents('https://avatars.githubusercontent.com/u/1836015')) === false) {
+            throw new Exception('image err1');
```

Comment:
Having to fetch the drawing from the web adds an extra dependency to this test, one that is certainly not needed for it. Please save the drawing as a local file, somewhere like `tests/data/Reader/HTML/memoryDrawingTest.jpg`. After that, it is not necessary to throw an exception when file_get_contents or imagecreatefromstring fails - you should just assert that the result of the function call is not false. Your test code should look something like:
```php
        $image = file_get_contents('tests/data/Reader/HTML/memoryDrawingTest.jpg');
        self::assertNotFalse($image, 'unable to read file');
        $image = imagecreatefromstring($image);
        self::assertNotFalse($image, 'unable to create image from string');
```

Reply:
I have done those, and thank u for point out the problems in my code.